p2p/src/peer/p2p_peer_reducer.rs

```diff
@@ -23,41 +23,41 @@ impl P2pPeerState {
                 // TODO: add bound to peers
                 let peer_state = p2p_state
                     .peers
-                    .entry(*peer_id)
+                    .entry(peer_id)
                     .or_insert_with(|| P2pPeerState {
                         is_libp2p: true,
-                        dial_opts: dial_opts.clone(),
+                        dial_opts: None,
                         identify: None,
                         status: P2pPeerStatus::Disconnected {
                             time: Timestamp::ZERO,
                         },
                     });
 
                 if let Some(dial_opts) = dial_opts {
-                    peer_state.dial_opts.get_or_insert(dial_opts.clone());
+                    let _ = peer_state.dial_opts.insert(dial_opts);
```

Comment:
I am confused about what is going on here (more about the old version than the new one).

Reply:
I have changed logic in this one, in old one `dial_opts` would only be current if peer didn't exist in peers already or if it didn't have `dial_opts`